File: snippets/dart_snippets/lib/connecting_lsp.dart
import 'package:dart_snippets/sdk_instance.dart';
import 'package:breez_sdk/bridge_generated.dart';

Future<void> getLspInfo() async {
  // ANCHOR: get-lsp-info
  String? lspId = await breezSDK.lspId();
  LspInformation? lspInfo = await breezSDK.lspInfo();
  print(lspId);
  print(lspInfo);
  // ANCHOR_END: get-lsp-info
}

Future<List<LspInformation>> listLsps() async {
  // ANCHOR: list-lsps
  List<LspInformation> availableLsps = await breezSDK.listLsps();
  // ANCHOR_END: list-lsps
  return availableLsps;
}

Future<void> connectLsp(String lspId) async {
  // ANCHOR: connect-lsp
  await breezSDK.connectLSP(lspId);
  // ANCHOR_END: connect-lsp
}

File: snippets/swift/BreezSDKExamples/Sources/ConnectingLsp.swift
//
//  ConnectingLsp.swift
//
//
//  Created by ruben on 14/11/2023.
//

import BreezSDK
import Foundation

func getLspInfo(sdk: BlockingBreezServices) -> LspInformation? {
    // ANCHOR: get-lsp-info
    let lspId = try? sdk.lspId()
    let lspInfo = try? sdk.lspInfo()
    // ANCHOR_END: get-lsp-info
    return lspInfo
}

func listLsps(sdk: BlockingBreezServices) -> [LspInformation]? {
    // ANCHOR: list-lsps
    let availableLsps = try? sdk.listLsps()
    // ANCHOR_END: list-lsps
    return availableLsps
}

func connectLsp(sdk: BlockingBreezServices, lspId: String) {
    // ANCHOR: connect-lsp
    try? sdk.connectLsp(lspId: lspId)
    // ANCHOR_END: connect-lsp
}
